crates/services/gas_price_service/src/ports.rs

```diff
@@ -30,25 +34,65 @@ pub trait GasPriceData: Send + Sync {
     fn latest_height(&self) -> Option<BlockHeight>;
 }
 
-pub struct GasPriceServiceConfig {
-    pub min_gas_price: u64,
-    pub starting_gas_price: u64,
-    pub gas_price_change_percent: u64,
-    pub gas_price_threshold_percent: u64,
+pub enum VersionSpecificConfig {
```

Comment:
Nit: I'd prefer the old name here `GasPriceServiceConfig`. The fact that it's versioned is clear enough given the `V0` and `V1` variants, and any code which doesn't match on the version variants shouldn't have to care if it's versioned or not.

```suggestion
pub enum GasPriceServiceConfig
```